_pytest/mark.py

```diff
@@ -4,17 +4,61 @@
 from operator import attrgetter
 from .compat import imap
 
+
 def alias(name):
     return property(attrgetter(name), doc='alias for ' + name)
 
 
+class ParameterSet(namedtuple('ParameterSet', 'values, marks, id')):
+    @classmethod
+    def param(cls, *value, **kw):
+        marks = kw.pop('marks', ())
+        
+        if isinstance(marks, MarkDecorator):
+            marks = marks,
+        else:
+            assert isinstance(marks, (tuple, list, set))
+
+        id = kw.pop('id', None)
+        assert not kw, kw
+        return cls(value, marks, id)
+
+    @classmethod
+    def extract_from(cls, param, legacy_force_tuple=False):
```

Comment:
will do, it may need a better name, its for the special case of single name parametrizations